ppinot-model/src/main/java/es/us/isa/ppinot/evaluation/computers/TimeMeasureComputer.java

```diff
@@ -257,50 +254,68 @@ public void update(LogEntry entry, boolean startMatches, boolean endMatches) {
         }
 
     }
-
     private class LinearMeasureInstanceTimer extends MeasureInstanceTimer {
 
         private DateTime start;
         private DurationWithExclusion duration;
+        private boolean stopped = false;
 
         public LinearMeasureInstanceTimer(TimeMeasure definition, String processId, String instanceId) {
             super(definition, processId, instanceId);
         }
 
         public void starts(DateTime start, LogEntry entry) {
-            if (! isRunning())
+            if (!isRunning()) {
                 this.start = start;
+            }
         }
 
         @Override
         protected double computeValue() {
             double value;
 
-            if (duration != null)
+            if (duration != null) {
                 value = duration.getMillis();
-            else {
-                if (isRunning() && !isProcessFinished() && ((TimeMeasure) definition).isComputeUnfinished()) {
-                    value = new DurationWithExclusion(start, now, ((TimeMeasure)definition).getConsiderOnly()).getMillis();
-                } else {
-                    value = Double.NaN;
-                }
+            } else if (isRunning() && !isProcessFinished() && ((TimeMeasure) definition).isComputeUnfinished()) {
+                value = new DurationWithExclusion(start, now, ((TimeMeasure) definition).getConsiderOnly()).getMillis();
+            } else {
+                value = Double.NaN;
             }
 
             return value;
         }
 
         public void ends(DateTime end) {
             if (isRunning()) {
-                duration = new DurationWithExclusion(start, end, ((TimeMeasure)definition).getConsiderOnly());
+                if ((this.isFirstTo() && !this.getStopped())) {
```

Comment:
Try to avoid duplicating the assignment of duration. I would do:

```
if (this.isRunning() && ! this.getStopped()) {
   duration = new DurationWithExclusion(start, end, ((TimeMeasure) definition).getConsiderOnly());
   if (this.isFirstTo()) {
     this.setStopped(true);
   }
} 
```

Please check that it works. I haven't tested it.

Reply:
That is true, thanks. I have just pushed the change https://github.com/isa-group/ppinot/pull/12/commits/96c2769cb0f980c8f1d6e66abc63c4c717512b2e.